community/kernel/src/main/java/org/neo4j/kernel/impl/newapi/RelationshipCursor.java

```diff
@@ -28,7 +28,8 @@
 abstract class RelationshipCursor extends RelationshipRecord implements RelationshipDataAccessor, RelationshipVisitor<RuntimeException>
 {
     Read read;
-    private HasChanges hasChanges = HasChanges.MAYBE;
+    private boolean hasChanges;
```

Comment:
An alternative, if you want to keep the original design, would have been to let `hasChanges` be an `int` and `MAYBE`, `YES, and `NO` be int constants. Either one is fine I'd guess.

Question out of interest: did the overhead of the `enum` actually make a dent in the performance, showing up in profiling?

Reply:
Yeah I considered the int alternative, but guessed this would be faster in the happy path. 

It was pretty surprising, but before this change `hasChanges()` actually showed measurable self time in the LDBC Core read benchmark! I always though enum comparison should be superfast, as they are essentially just highly limited integers. This fix seems to remove that self time. Maybe that profile was misleading, but I though Async was pretty accurate.